packages/api/src/controllers/taskController.js

```diff
@@ -245,14 +247,42 @@ const taskController = {
     }
   },
 
+  async checkCustomDependencies(typeOfTask, data, farm_id) {
```

Comment:
https://vincit.github.io/objection.js/api/query-builder/mutate-methods.html#upsertgraph
Documentation for upsert graph. These are already implemented by objection js. Just add proper relation mappings to task models and validations